src/polytopes/chain.jl

```diff
@@ -4,18 +4,34 @@
 
 """
     Chain(p1, p2, ..., pn)
+    Chain(points)
 
-A polygonal chain from a sequence of points `p1`, `p2`, ..., `pn`.
+A polygonal chain from a sequence of points `p1`, `p2`, ..., `pn` or a vector
+of `points`. If the first point is equal to the last point or if `points` is of
+type `CircularVector`, the chain is considered closed.
 See https://en.wikipedia.org/wiki/Polygonal_chain.
 """
 struct Chain{Dim,T,V<:AbstractVector{Point{Dim,T}}} <: Polytope{1,Dim,T}
   vertices::V
+
+  function Chain(vertices::AbstractVector{Point{Dim,T}}) where {Dim,T}
+    if !isa(vertices, CircularVector) && length(vertices) >= 2 &&
+        first(vertices) == last(vertices)
+      vertices = CircularVector(vertices[begin:end-1])
+    end
+    new{Dim,T,typeof(vertices)}(vertices)
+  end
```

Comment:
We need to implement this with multiple-dispatch at compile-time instead of checking the types if `isa`. This is not idiomatic Julia.

Reply:
I’m pretty sure `isa` is always resolved at compile-time (where possible) and it is used frequently in Base and throughout the Julia ecosystem. In this case I agree that it might be cleaner to rely on multiple but in general I think multiple dispatch makes sense where most/all of the functionality of a function is different and `isa` makes more sense when only a small part differs and that small part doesn’t make sense as a separate function.